### DISCUSSION TITLE
How to build only tsh without Docker?

### DISCUSSION THREAD

**Discussion by tuaris** [gravitational/teleport#38808] [Q&A]
I would like to build only `tsh`, but without using Docker since that doesn't work on my UNIX platform.  Using the command `make build/tsh` complains about not being able to find Kubernetes dependencies.

**Comment by stevenGravy** [comment]
Which unix platform is this?

**Reply by tuaris** [reply]
I'm on FreeBSD 13.2.  For the moment I am using the built in Linux Emulation layer to work around this problem.  It would be nice though to have native support even if just for the `tsh` client since FreeBSD has no issues running Golang applications.

**Reply by webvictim** [reply]
Unfortunately Teleport has some CGO/Rust dependencies which will make it more challenging to compile natively on FreeBSD.

**Reply by tuaris** [reply]
Yep, I get that.  This is my personal project to update the (very out of date) FreeBSD port.

**Comment by webvictim** [comment]
`make build/tsh` should work. Can you share the exact error?

**Reply by tuaris** [reply]
I did more troubleshooting since I opened this and I'm going to give it another go after updating to a newer version of golang.  If it still fails I'll post the results here.
